src/cobra/core/reaction.py

```diff
@@ -988,37 +1111,40 @@ def format(number):
         reaction_string += " + ".join(product_bits)
         return reaction_string
 
-    def check_mass_balance(self):
-        """Compute mass and charge balance for the reaction
+    def check_mass_balance(self) -> Dict[str, Union[int, float]]:
+        """Compute mass and charge balance for the reaction.
 
-        returns a dict of {element: amount} for unbalanced elements.
-        "charge" is treated as an element in this dict
-        This should be empty for balanced reactions.
+        Returns
+        -------
+        dict
+            a dict of {element: amount} for unbalanced elements.
+            "charge" is treated as an element in this dict
+            This should be empty for balanced reactions.
         """
         reaction_element_dict = defaultdict(int)
         for metabolite, coefficient in self._metabolites.items():
             if metabolite.charge is not None:
                 reaction_element_dict["charge"] += coefficient * metabolite.charge
             if metabolite.elements is None:
-                raise ValueError("No elements found in metabolite %s" % metabolite.id)
+                raise ValueError(f"No elements found in metabolite {metabolite.id}")
             for element, amount in metabolite.elements.items():
                 reaction_element_dict[element] += coefficient * amount
         # filter out 0 values
         return {k: v for k, v in reaction_element_dict.items() if v != 0}
 
     @property
-    def compartments(self):
-        """lists compartments the metabolites are in"""
+    def compartments(self) -> Set:
+        """Return set of compartments the metabolites are in."""
         return {
             met.compartment for met in self._metabolites if met.compartment is not None
         }
 
-    def get_compartments(self):
-        """lists compartments the metabolites are in"""
+    def get_compartments(self) -> list:
```

Comment:
Maybe add a Returns section to the docstring? Can be kept as it is as it has a deprecation warning on it.